spring-boot-starter/spring-boot-starter-discovery/src/main/java/io/servicecomb/springboot/starter/serviceregistry/CseServiceRegistry.java

```diff
@@ -20,21 +20,23 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import io.servicecomb.serviceregistry.api.registry.Microservice;
+import io.servicecomb.serviceregistry.client.RegistryClientFactory;
 import io.servicecomb.serviceregistry.client.ServiceRegistryClient;
 
 public class CseServiceRegistry implements ServiceRegistry<CseRegistration> {
 
 	private static final Logger log = LoggerFactory.getLogger(CseServiceRegistry.class);
 
-	private final ServiceRegistryClient client;
+	private ServiceRegistryClient client;
 
-	public CseServiceRegistry(ServiceRegistryClient client) {
-		this.client = client;
-		try {
-			this.client.init();
-		} catch (Exception e) {
-			// TODO Auto-generated catch block
-			e.printStackTrace();
+	public CseServiceRegistry() {
+		if (null == this.client) {
+			this.client = RegistryClientFactory.getRegistryClient();
+			try {
+				this.client.init();
```

Comment:
please rebase onto branch JAV-30_discovery_client. this method shall not throw any exception.